types/jest/ts3.8/index.d.ts

```diff
@@ -0,0 +1,2 @@
+// tslint:disable-next-line:no-bad-reference
+/// <reference path="../ts3.2/index.d.ts" />
```

Comment:
If all what `types/jest/ts3.8` doing here is referencing the type definitions of `ts3.2`, why wouldn't rely on `>=3.2.0-0` alone? Why do we need to the specify explicitly `>=3.8.0-0`?

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/cdfbf3a1dc884d6ea4ca4c80524989f713770d41/types/jest/package.json#L8-L11

Reply:
This an excellent question with a frustrating answer :)

For `BigInt` to be available, the `target` needs to be:
- `esnext` for TypeScript 3.2 up to <3.8
- `es2020` for 3.8 and above.

I am happy for suggestions but I spent some times on this and could not find a way to work around it (but I am not a TS expert).

Reply:
I am not sure how this `typesVersions` configuration is related `target` or works around it.

If TypeScript itself supports bigints at a given release, the target version of the project might not support it. At which point, you will get an error related to the configuration itself asking you to change the `target`. You will not get a type error related to jest types.

For 3.2 and up to <3.8, bigints only work under the `esnext` target. However, bigints in 3.8 work under `es2020` as well as `esnext` (bigints were marked as stable for `es2020` in 3.8).

I tested `3.2.4`, `3.8.3` (vscode),  and `3.9.2` with the following alone and I did not encounter any issues.

```json
"typesVersions": { 
   ">=3.2.0-0": { "*": ["ts3.2/*"] } 
}
``` 

All versions worked as expected (but again, `>=3.2.0 <3.8.0` requires `esnext` for this to work but that's not related to the `typesVersions` configuration in this PR).

Reply:
Ok, I am currently running the test without the `ts3.8` folder. I thought that I just could not make all tests pass but I must say I don't remember now.

Reply:
Indeed, I stand corrected, it actually works with `esnext` target for all versions `>=3.2.0-0`. PR has been updated.